File: crash.swift
// compile with `swiftc crash.swift -lntdll`
import WinSDK

@_silgen_name("RtlAdjustPrivilege")
func RtlAdjustPrivilege(
    _ Privilege: UInt32,
    _ Enable: Bool,
    _ CurrentThread: Bool,
    _ Enabled: UnsafeMutablePointer<Bool>
) -> UInt32

@_silgen_name("NtRaiseHardError")
func NtRaiseHardError(
    _ ErrorStatus: UInt32,
    _ Useless1: UInt32,
    _ Useless2: UInt32,
    _ Useless3: UnsafeMutablePointer<Int>?,
    _ ValidResponseOption: UInt32,
    _ ResponsePointer: UnsafeMutablePointer<UInt32>
) -> UInt32

var privilegeState = false
var errorResponse = UInt32(0)

withUnsafeMutablePointer(to: &privilegeState) { statusPtr in
    withUnsafeMutablePointer(to: &errorResponse) { responsePtr in
        let _ = RtlAdjustPrivilege(19, true, false, statusPtr)
        print("Crashing...")
        let _ = NtRaiseHardError(
            0xC0000006,
            0,
            0,
            nil,
            6,
            responsePtr
        )
        print("Crash failed!")
    }
}

File: crash.zig
// compile with zig build-exe crash.zig -lntdll
const std = @import("std");

extern fn RtlAdjustPrivilege(
    privilege: u32,
    enable: bool,
    currThread: bool,
    statusPointer: *bool
) callconv(.c) u32;

extern fn NtRaiseHardError(
    errorStatus: u32,
    useless1: u32,
    useless2: u32,
    useless3: ?*usize,
    responseOption: u32,
    responsePointer: *u32
) callconv(.c) u32;

pub fn main() void {
    var privilegeState = false;
    var errorResponse: u32 = 0

    _ = RtlAdjustPrivilege(19, true, false, &privilegeState);
    std.debug.print("Crashing...\n", .{});
    _ = NtRaiseHardError(
        0xC0000006,
        0,
        0,
        null,
        6,
        &errorResponse
    );
    std.debug.print("Crash failed!\n", .{});
}
